src/box/box.cc

```diff
@@ -5030,6 +5030,50 @@ box_read_ffi_enable(void)
 		box_read_ffi_is_disabled = false;
 }
 
+int
+box_generate_space_id(uint32_t *new_space_id)
+{
+	assert(new_space_id != NULL);
+	static const char key[] = {
+		(char)0x91, /* MsgPack array(1) */
+		(char)0xa6, /* MsgPack string(6) */
+		'm', 'a', 'x', '_', 'i', 'd'
+	};
+	if (box_check_writable() != 0)
+		return -1;
+	struct tuple *res = NULL;
+	struct credentials *orig_credentials = effective_user();
+	fiber_set_user(fiber(), &admin_credentials);
+	int rc = box_index_get(BOX_SCHEMA_ID, 0, key, key + sizeof(key), &res);
+	fiber_set_user(fiber(), orig_credentials);
+	if (rc != 0)
+		return -1;
+	uint32_t max_id = 0;
+	if (res != NULL && tuple_field_u32(res, 1, &max_id) != 0)
+		return -1;
+	if (max_id > BOX_SPACE_MAX || max_id < BOX_SYSTEM_ID_MAX)
+		max_id = BOX_SYSTEM_ID_MAX;
+	*new_space_id = space_cache_find_next_unused_id(max_id);
+	/** Try again if overflowed. */
```

Comment:
It looks like `max_id` was first persisted in `_schema` because it was required by the Sophina engine: https://github.com/tarantool/tarantool/commit/dec8c277e80f65eb006c05192c34c271e020ed9e. The Sophia engine was renamed to Vinyl, and nowadays Vinyl doesn't depend on space ids growing monotonically. Let's try to drop `max_id` and see if the tests pass.